pandas/tests/frame/test_analytics.py

```diff
@@ -1075,6 +1075,29 @@ def test_mean_excludeds_datetimes(self, tz):
         expected = pd.Series()
         tm.assert_series_equal(result, expected)
 
+    def test_mean_mixed_string_float(self):
+        # GH 11670
+        # possible bug when calculating mean of DataFrame?
+
+        d = [
+            {"A": 2, "B": None, "C": 628.00},
```

Comment:
The `C` values need to be `Decimal` objects. Could you also update the test name to reflect that?